src/operations.rs

```diff
@@ -110,6 +111,9 @@ impl SwapOperationsListUnchecked {
     }
 
     pub fn check(&self, api: &dyn Api) -> Result<SwapOperationsList, ContractError> {
+        let mut assets_to_swap = HashSet::new();
+        // should not visit the same asset more than once
+        let mut visited_assets = HashSet::new();
```

Comment:
i feel like this might be a too restrictive constraint. In this use case it is valid but i figured we might want to use this struct for more general swaps at some point. we could not use if for arbs for example if we restrict this way

Thoughts? @apollo-sturdy  @apolloshab